src/lib/common/string.cpp

```diff
@@ -676,3 +676,45 @@ void strReplace(char* to, int toLen, const char* from, const char* oldString, co
 
   to[toIx] = 0;
 }
+
+
+
+/* ****************************************************************************
+*
+* servicePathCheck - check one single component of the service-path
+*
+* NOTE
+* This function is used for register/notify only. '#' is not an accepted character,
```

Comment:
From a functional point of view, the constrain that this function enforces also applies to update, given that in that case `#` is not accepted. I understand that adapting the code to use also this function for updates is what the "FIXME P5: updates should also call the other servicePathCheck (in common lib)" in other place of the code is about, right?